File: benchmarks/language/sequence_io/r/gc_content_scale.R
library(Biostrings)
records <- readDNAStringSet("data_real/human_chr22.fa")
gc_values <- letterFrequency(records, letters="GC", as.prob=TRUE)[,1]
cat(sprintf("Sequences: %d\n", length(records)))
cat(sprintf("Mean GC: %.4f\n", mean(gc_values)))
cat(sprintf("Min GC: %.4f\n", min(gc_values)))
cat(sprintf("Max GC: %.4f\n", max(gc_values)))

File: benchmarks/language/sequence_io/python/gc_content_scale.py
"""GC Content at Scale: compute GC for every sequence in a large FASTA"""
from Bio import SeqIO
from Bio.SeqUtils import gc_fraction

records = list(SeqIO.parse("data_real/human_chr22.fa", "fasta"))
gc_values = [gc_fraction(r.seq) for r in records]
print(f"Sequences: {len(records)}")
print(f"Mean GC: {round(sum(gc_values) / len(gc_values), 4)}")
print(f"Min GC: {round(min(gc_values), 4)}")
print(f"Max GC: {round(max(gc_values), 4)}")
